cantor-s3/src/main/java/com/salesforce/cantor/s3/ObjectsOnS3.java

```diff
@@ -223,16 +230,21 @@ private void doDrop(final String namespace) throws IOException {
         this.s3Client.deleteBucket(bucket);
 
         // check bucket deleted successfully
+        // todo: remove this check? race condition exists where current thread deletes a bucket and another thread creates a bucket right after, causing this to throw an exception
```

Comment:
I think that would be a rare race condition, it's ok to throw an exception in that case, so let's keep this.